mocks/tests/test_mocks.py

```diff
@@ -118,8 +122,9 @@ def test_parse_config_kwargs():
     assert config['m_nu'].unit == u.eV
 
 
-def test_read_gadget(gadget_data_path):
-    pos, mass, info = mocks.read_gadget(gadget_data_path)
+def test_read_gadget():
+    config = mocks.parse_config()
```

Comment:
maybe instead of calling parse_config() here you should define it as a fixture in in conftest.py? You don't want an error in parse_config() to cause the test_read_gadget() to fail.

Reply:
@fergusL 
The `parse_config` function is tested before this test (check lines 46 and 76), so if there is an error in it, it will fail before this test. However, I do not know if it is a good reason or not. What do you think with respect to what I mentioned? Should I define a fixture for that?

Reply:
@fergusL 
The `parse_config` function is tested before this test (check lines 46 and 76), so if there is an error in it, it will fail before this test. However, I do not know if it is a good reason or not. What do you think with respect to what I mentioned? Should I define a fixture for it and use it in `test_read_gadget`?

Reply:
I think it probably doesn't matter, maybe if you think other tests down the line will also use the config from parse_config()? Then it'd make sense to just set it as a fixture in conftest.py, so youre not recreating the object each time its needed in a test.